### PR TITLE
runelite-client: code cleanup (and clue scroll fix)

### PR DESCRIPTION
**Pull request by sdburns1998**

### CODE REVIEW

File: runelite-client/src/main/java/net/runelite/client/plugins/cannon/CannonPlugin.java [hunk 1]
```diff
@@ -250,17 +248,15 @@ public void onProjectileMoved(ProjectileMoved event)
 			WorldPoint projectileLoc = WorldPoint.fromLocal(client, projectile.getX1(), projectile.getY1(), client.getPlane());
 
 			//Check to see if projectile x,y is 0 else it will continuously decrease while ball is flying.
-			if (projectileLoc.equals(cannonPosition) && projectile.getX() == 0 && projectile.getY() == 0)
-			{
+			if (projectileLoc.equals(cannonPosition) && projectile.getX() == 0 && projectile.getY() == 0 &&
```
Comment (Ganom) [inline runelite-client/src/main/java/net/runelite/client/plugins/cannon/CannonPlugin.java]: Dont really need to include the comment in this lol.

File: runelite-client/src/main/java/net/runelite/client/plugins/chathistory/ChatHistoryPlugin.java [hunk 2]
```diff
@@ -143,13 +143,11 @@ public void onChatMessage(ChatMessage chatMessage)
 			case MODPRIVATECHAT:
 				final String name = Text.removeTags(chatMessage.getName());
 				// Remove to ensure uniqueness & its place in history
-				if (!friends.remove(name))
-				{
+				if (!friends.remove(name) &&
 					// If the friend didn't previously exist ensure deque capacity doesn't increase by adding them
```
Comment (Ganom) [inline runelite-client/src/main/java/net/runelite/client/plugins/chathistory/ChatHistoryPlugin.java]: No need to include comment here either.

File: runelite-client/src/main/java/net/runelite/client/plugins/banlist/BanListPlugin.java [hunk 3]
```diff
@@ -121,22 +122,19 @@ protected void shutDown() throws Exception
 	@Subscribe
 	public void onConfigChanged(ConfigChanged event)
 	{
-		if (event.getGroup().equals("banlist"))
+		if (event.getGroup().equals("banlist") && event.getKey().equals("bannedPlayers"))
 		{
-			if (event.getKey().equals("bannedPlayers"))
+			for (String manual : Text.fromCSV(config.getBannedPlayers()))
```
Comment (Lucwousin) [inline runelite-client/src/main/java/net/runelite/client/plugins/banlist/BanListPlugin.java]: Won't this cause any issues if the string is empty? Also keep in mind you're doing contains before standardizing, so it would say no, doesn't contain while it actually may

File: runelite-client/src/main/java/net/runelite/client/plugins/banlist/BanListPlugin.java [hunk 4]
```diff
@@ -201,24 +199,21 @@ public void onClanMemberJoined(ClanMemberJoined event)
 	@Subscribe
 	public void onWidgetLoaded(WidgetLoaded widgetLoaded)
 	{
-		if (this.highlightInTrade)
-		{
-			if (widgetLoaded.getGroupId() == 335)
-			{ //if trading window was loaded
-				clientThread.invokeLater(() ->
+		if (this.highlightInTrade && widgetLoaded.getGroupId() == 335)
```
Comment (Lucwousin) [inline runelite-client/src/main/java/net/runelite/client/plugins/banlist/BanListPlugin.java]: Can we use WidgetID instead of magic numbers pl0x

File: runelite-client/src/main/java/net/runelite/client/plugins/banlist/BanListPlugin.java [hunk 5]
```diff
@@ -227,28 +222,19 @@ public void onWidgetLoaded(WidgetLoaded widgetLoaded)
 	 */
 	private ListType checkScamList(String nameToBeChecked)
 	{
-		if (wdrScamArrayList.size() > 0 && this.enableWDR)
+		if (wdrScamArrayList.size() > 0 && this.enableWDR && wdrScamArrayList.stream().anyMatch(nameToBeChecked::equalsIgnoreCase))
```
Comment (Lucwousin) [inline runelite-client/src/main/java/net/runelite/client/plugins/banlist/BanListPlugin.java]: The name collections should be HashSets, so you can just use .contains() to look them up as quick as you can

File: runelite-client/src/main/java/net/runelite/client/plugins/barrows/BarrowsOverlay.java [hunk 6]
```diff
@@ -97,7 +97,7 @@ public Dimension render(Graphics2D graphics)
 			final List<Player> players = client.getPlayers();
 			for (Player player : players)
 			{
-				if (player == local)
+				if (player.equals(local))
```
Comment (Lucwousin) [inline runelite-client/src/main/java/net/runelite/client/plugins/barrows/BarrowsOverlay.java]: `==` was correct here

File: runelite-client/src/main/java/net/runelite/client/plugins/chatcommands/ChatCommandsPlugin.java [hunk 7]
```diff
@@ -818,7 +818,7 @@ private void itemPriceLookup(ChatMessage chatMessage, String message)
 			}
 			catch (IOException e)
 			{
-				e.printStackTrace();
+				log.error(e.toString());
```
Comment (Lucwousin) [inline runelite-client/src/main/java/net/runelite/client/plugins/chatcommands/ChatCommandsPlugin.java]: `log.error("Error looking up prices", e);` would work better imo

File: runelite-client/src/main/java/net/runelite/client/plugins/chattranslation/ChatTranslationPlugin.java [hunk 8]
```diff
@@ -24,26 +35,22 @@
...
+@Slf4j
 public class ChatTranslationPlugin extends Plugin implements KeyListener
 {
 
 	private static final String TRANSLATE = "Translate";
 
 	private static final ImmutableList<String> AFTER_OPTIONS = ImmutableList.of("Message", "Add ignore", "Remove friend", "Kick");
 
-	private ArrayList<String> playerNames = new ArrayList<>();
+	private final List<String> playerNames = new ArrayList<>();
```
Comment (Lucwousin) [inline runelite-client/src/main/java/net/runelite/client/plugins/chattranslation/ChatTranslationPlugin.java]: This should be a set

File: runelite-client/src/main/java/net/runelite/client/plugins/chattranslation/ChatTranslationPlugin.java [hunk 9]
```diff
@@ -223,7 +222,7 @@ public void onChatMessage(ChatMessage chatMessage)
 				}
 				catch (Exception e)
 				{
-					e.printStackTrace();
+					log.warn(e.toString());
```
Comment (Lucwousin) [inline runelite-client/src/main/java/net/runelite/client/plugins/chattranslation/ChatTranslationPlugin.java]: `log.warn("${some message}", e);`

File: runelite-client/src/main/java/net/runelite/client/plugins/chattranslation/ChatTranslationPlugin.java [hunk 10]
```diff
@@ -246,47 +245,42 @@ public void keyPressed(KeyEvent event)
...
-							client.runScript(96, 0, translation);
-						});
-					}
-					client.setVar(VarClientStr.CHATBOX_TYPED_TEXT, "");
-				}
-				catch (Exception e)
-				{
-					e.printStackTrace();
+					clientThread.invoke(() ->
+						client.runScript(96, 0, translation));
```
Comment (Lucwousin) [inline runelite-client/src/main/java/net/runelite/client/plugins/chattranslation/ChatTranslationPlugin.java]: Can we add 96 to ScriptID as well

File: runelite-client/src/main/java/net/runelite/client/plugins/clanchat/ClanChatPlugin.java [hunk 11]
```diff
@@ -197,7 +197,7 @@ public void onClanMemberJoined(ClanMemberJoined event)
 
 			for (final Player player : client.getPlayers())
 			{
-				if (player != null && player != local && memberName.equals(Text.toJagexName(player.getName())))
+				if (player != null && !player.equals(local) && memberName.equals(Text.toJagexName(player.getName())))
```
Comment (Lucwousin) [inline runelite-client/src/main/java/net/runelite/client/plugins/clanchat/ClanChatPlugin.java]: == again but you probably already knew that by now

File: runelite-client/src/main/java/net/runelite/client/plugins/combatcounter/CombatCounter.java [hunk 12]
```diff
@@ -114,7 +116,7 @@ CombatCounterConfig provideConfig(ConfigManager configManager)
 		return configManager.getConfig(CombatCounterConfig.class);
 	}
 
-	private Map<Integer, Integer> variables = new HashMap<Integer, Integer>()
+	private final Map<Integer, Integer> VARIABLES = new HashMap<Integer, Integer>()
```
Comment (Lucwousin) [inline runelite-client/src/main/java/net/runelite/client/plugins/combatcounter/CombatCounter.java]: This looks like it should be a static final ImmutableMap (and the data in it looks like it should go into api instead of here)

File: runelite-client/src/main/java/net/runelite/client/plugins/combatcounter/CombatCounter.java [hunk 13]
```diff
@@ -230,7 +232,7 @@ CombatCounterConfig provideConfig(ConfigManager configManager)
 		}
 	};
 
-	private List<Integer> MAGE_ANIMATIONS = new ArrayList<Integer>()
+	private final List<Integer> MAGE_ANIMATIONS = new ArrayList<Integer>()
```
Comment (Lucwousin) [inline runelite-client/src/main/java/net/runelite/client/plugins/combatcounter/CombatCounter.java]: All *_ANIMATIONS lists should be static immutablesets

File: runelite-client/src/main/java/net/runelite/client/plugins/combatcounter/CombatCounter.java [hunk 14]
```diff
@@ -316,7 +320,7 @@ public void onAnimationChanged(AnimationChanged event)
 								for (NPC nearby : this.client.getNpcs())
 								{
 									int distance = npc.getWorldLocation().distanceTo(nearby.getWorldLocation());
-									if (distance <= 1 && npc != nearby)
+									if (distance <= 1 && npc.equals(nearby))
```
Comment (Lucwousin) [inline runelite-client/src/main/java/net/runelite/client/plugins/combatcounter/CombatCounter.java]: Is this change on purpose (!= would work here, so would == but I don't know if youre supposed to be checking equality or inequality)

File: runelite-client/src/main/java/net/runelite/client/plugins/corp/CorpPlugin.java [hunk 15]
```diff
@@ -224,7 +224,7 @@ public void onInteractingChanged(InteractingChanged interactingChanged)
 		Actor source = interactingChanged.getSource();
 		Actor target = interactingChanged.getTarget();
 
-		if (corp == null || target != corp)
+		if (target != null && !target.equals(corp))
```
Comment (Lucwousin) [inline runelite-client/src/main/java/net/runelite/client/plugins/corp/CorpPlugin.java]: this can all be ==/!= as well

File: runelite-client/src/main/java/net/runelite/client/plugins/itemprices/ItemPricesOverlay.java [hunk 16]
```diff
@@ -97,7 +97,10 @@ public Dimension render(Graphics2D graphics)
 		switch (action)
 		{
 			case ITEM_USE_ON_WIDGET:
-				if (!menuEntry.getTarget().contains("High Level Alchemy") || !plugin.isShowAlchProfit()) return null;
+				if (!menuEntry.getTarget().contains("High Level Alchemy") || !plugin.isShowAlchProfit())
```
Comment (Lucwousin) [inline runelite-client/src/main/java/net/runelite/client/plugins/itemprices/ItemPricesOverlay.java]: client.getSelectedSpellName() can be used for this

File: runelite-client/src/main/java/net/runelite/client/plugins/lizardmenshaman/LizardmenShamanPlugin.java [hunk 17]
```diff
@@ -102,12 +102,9 @@ protected void shutDown() throws Exception
 	@Subscribe
 	public void onChatMessage(ChatMessage event)
 	{
-		if (this.notifyOnSpawn)
+		if (this.notifyOnSpawn && event.getMessage().contains(MESSAGE))
```
Comment (Lucwousin) [inline runelite-client/src/main/java/net/runelite/client/plugins/lizardmenshaman/LizardmenShamanPlugin.java]: I think checking messagetype as well may be a bit faster, if you do that before checking the message

File: runelite-client/src/main/java/net/runelite/client/plugins/lizardmenshaman/LizardmenShamanPlugin.java [hunk 18]
```diff
@@ -120,12 +117,9 @@ public void onAnimationChanged(AnimationChanged event)
 			return;
 		}
 
-		if (actor.getName().equals(SHAMAN) && actor.getAnimation() == 7157)
+		if (actor.getName().equals(SHAMAN) && actor.getAnimation() == 7157 && this.showTimer)
```
Comment (Lucwousin) [inline runelite-client/src/main/java/net/runelite/client/plugins/lizardmenshaman/LizardmenShamanPlugin.java]: AnimationID.class pl0x

File: runelite-client/src/main/java/net/runelite/client/plugins/lizardmenshaman/LizardmenShamanPlugin.java (hunk 17 above)
Comment (Owain94) [reply runelite-client/src/main/java/net/runelite/client/plugins/lizardmenshaman/LizardmenShamanPlugin.java]: Marking as reminder
- Don't know the message type

Comment (Kyleeld) [reply runelite-client/src/main/java/net/runelite/client/plugins/lizardmenshaman/LizardmenShamanPlugin.java]: chatMessage.getType() == ChatMessageType.GAMEMESSAGE)
